src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/MSBuild.cs

```diff
@@ -314,14 +314,14 @@ public async Task<bool> ExecuteInternal()
             {
                 ITaskItem project = Projects[i];
 
-                string projectPath = FileUtilities.AttemptToShortenPath(project.ItemSpec);
+                string projectPath = FileUtilities.GetFullPathNoThrow(project.ItemSpec);
```

Comment:
It seems like we shorten the path always after this change. I guess the explanation for the change is "So, when project path includes multiple reductant "..\\" not longer than the max path, it will report MSBuild.ProjectFileNotFound.". I am a bit confused: if the path is not too long, how "MSBuild.ProjectFileNotFound" is thrown? I would think that the file would be found in this case.

Reply:
When project path includes multiple reductant "..\" not longer than the max path. The code here `FileSystems.Default.FileExists(projectPath) `will return false.
https://github.com/dotnet/msbuild/blob/b59f07e4312eb6e3e33e59241453606c81992738/src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/MSBuild.cs#L343-L392

Reply:
Is this discussion still relevant?

Reply:
It's related. But it has been resolved by our new changes